### ISSUE TITLE
CE dates between 1 and 31 are invalid in DateRange datatype, 32-99 converted to 1932, etc.

### ISSUE THREAD

**Issue by ewg118** [collectiveaccess/providence#867]
I am not sure why there seems to be an arbitrary division between 31 and 32, but when the DateRange type (for ISO 8601) is set on a field for first century CE dates, the numbers 1-31 have an invalid data warning. The numbers 32 to 99 are changed to 1932 and 1999, and so forth. BCE dates parse correctly. I have not tested the data export to know whether these are converted 4 digit numbers with padded zeroes, however. Does the date 100 get converted to '0100' on export into machine readable data?

**Comment by collectiveaccess** [comment]
It's probably because we window two digit years by default. We'll need to look at that.

**Comment by ewg118** [comment]
I wanted to follow up on this since I've discovered some new formatting issues with the year. According to ISO 8601, the year should always be 4 digits, with padded 0s. So 14 BC is -0014. If I set `{ "start_as_iso8601" : 1 }` in my exporter, it comes through as `-14-01-01`.

There's actually an issue between ISO 8601 and the date standard for the XML schema datatypes. In ISO 8601, 1 BC is '0000', so that means 14 BC is '-0013'. In XSD 1.0, upon which most Java software implementations are built (e.g., XML applications, SPARQL endpoints, etc.), 14 BC is '-0014'. Javascript libraries follow ISO 8601, not XSD. So it may be useful to introduce an option in the exporter to designate the format as ISO 8601 or XSD, if developers need to create XML or JSON.
